common/arch/sdl/window.cpp

```diff
@@ -72,21 +92,19 @@ int window_close(window *wind)
 		return 0;
 	}
 
-	if (wind == FrontWindow)
-		FrontWindow = wind->prev;
-	if (wind == FirstWindow)
-		FirstWindow = wind->next;
-	if (wind->next)
-		wind->next->prev = wind->prev;
-	if (wind->prev)
-		wind->prev->next = wind->next;
+	if (type == window_alloc_type::separate)
+		wind->unlink();		// Remove from the window list
```

Comment:
Yes, this was a very tricky one to resolve. The reason I did it this way is because if the handler of the `EVENT_WINDOW_CLOSE` deletes a subclassed window, it deletes the window with it. Attempting to unlink it after it's freed would cause a bad memory access. It can only know whether to unlink it if the handler returns `ignored`.

Yes, it would be a lot simpler for the handler to always delete it. I was avoiding this at first, but now doing it this way to avoid the potential issue you described.